### ISSUE TITLE
Ensure that CiviCRM contacts can be linked with Drupal authors

### ISSUE THREAD

**Issue by brylie** [WesternFriend/WF-website#462]
Since we switched to storing the Drupal author ID field instead of `full_name`, we may no longer be able to link CiviCRM contacts with the respective Drupal author. Therefore, we may need to add the `drupal_author_id` and `drupal_id` for the CiviCRM contact to be joined with the author during import.

## Task

- [ ] ensure that CiviCRM contacts can be linked with Drupal authors during the CiviCRM contact import
- [ ] verify whether to import clerk relationships or deprecate that feature

**Comment by brylie** [comment]
One data quality issue is that several CiviCRM contacts are missing the organization subtype. Without that subtype, we cannot import the organizations into the correct categories (meeting or organization subtypes).

**Comment by marycklein** [comment]
Hi Brylie -

I added contact type to all the contacts that I could. Here are notes of some oddities:

 “Unaffiliated ?” — As far as I could tell, this was connected with nothing, so I deleted the contact. However, it is still in the trash. It did not have any relationships with any other contacts before I deleted it.

“Library of the religious society of Friends” — has two listings as the type “individual” (which i ignored) and one listing as an “organization,” which I categorized

“Salt Lake Monthly Meeting” — my guess is that this contact may have been created when Charlene Weir made an online donation on behalf of the Meeting. The “correct” meeting contact, which is in our “directory of meetings” is “Salt Lake Friends Meeting.” I “deleted” the “Monthly Meeting” version of these two contacts. However, it is still in the trash.

“Phoenix Friends Meeting” — similar to Salt Lake; in this case, the canonical name in our directory is “Phoenix Monthly Meeting,” and a member of the meeting made an online payment on behalf of the meeting and used the slightly different name. I “deleted” the “Friends Meeting” version, but left the contact in the trash.

I both above two cases -- Salt Lake and Phoenix, I have records of the donations in multiple other sets of records — within our PayPal account, within our Transactions Record, within our monthly financials, and within our database of supporters. So we can afford to loose the data points in CiviCRM that these two payments were made in 2018 and 2019.

Two of these contacts, I categorized as “quaker organizations,” although they are simply “organization,” and not “quaker.” But we don’t have a sub-category for that at this point. Those were:
The Compassionate Listening Project
Arizona Theatre Matters

OK. That’s what I can do here for now.

See you tomorrow! (Tomorrow for me, anyway.)

Best,
Mary

Mary Klein
Editor / Executive Director

Western Friend
2625 Middlefield Road, #565
Palo Alto, CA  94306-2516
650-318-1652
